cmd/bbolt/main.go

```diff
@@ -1192,8 +1192,10 @@ func (cmd *benchCommand) runWritesWithSource(db *bolt.DB, options *BenchOptions,
 
 			fmt.Fprintf(cmd.Stderr, "Starting write iteration %d\n", i)
 			for j := int64(0); j < options.BatchSize; j++ {
-				key := make([]byte, options.KeySize)
-				value := make([]byte, options.ValueSize)
+				buf := make([]byte, options.KeySize+options.ValueSize)
+
+				key := buf[:options.KeySize:options.KeySize]
+				value := buf[options.KeySize:]
```

Comment:
I do not see much value of such change. The motivation might be reducing the number of memory allocation, but the existing implementation is simpler & clearer, accordingly better readability.

Reply:
Hi @ahrtr ,
thanks for your review. I must admit that this PR does add a bit of complexity.
However, I believe that the code still remains maintainable. The motivation to add this change is that lots of projects on the internet use this library and therefore will benefit of this optimisation.